# Phase 1: Dataset Overview

**The data used for this project has been obtained from Kaggle and belongs to Instacart.
Instacart is an online grocery and delivery app. Instacart posted this dataset as a part of a kaggle
competition 4 years ago. The dataset consists of 7 csv files, each file representing unique data.
The dataset is a relational dataset and the user data provided within the dataset is anonymized.
For the sake of this project, we will be using only 5 csv files.**

**Importing all the important libraries**

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from random import randint

**Importing all datasets**

In [2]:
df_aisle = pd.read_csv("aisles.csv")

df_dept = pd.read_csv("departments.csv")

df_prior = pd.read_csv("order_products__prior.csv")

df_order = pd.read_csv("orders.csv")

df_product = pd.read_csv("products.csv") 

In [3]:
print(df_aisle.shape)
df_aisle.head()

(134, 2)


aisle_id                       aisle
0         1       prepared soups salads
1         2           specialty cheeses
2         3         energy granola bars
3         4               instant foods
4         5  marinades meat preparation

In [4]:
print(df_dept.shape)
df_dept.head()

(21, 2)


department_id department
0              1     frozen
1              2      other
2              3     bakery
3              4    produce
4              5    alcohol

In [5]:
print(df_product.shape)
df_product.head()

(49688, 4)


product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  
0             19  
1             13  
2              7  
3              1  
4             13

In [6]:
print(df_prior.shape)
df_prior.head()

(32434489, 4)


order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [7]:
print(df_order.shape)
df_order.head()

(3421083, 7)


order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

**Understanding the Dataset**

1. aisles.csv: In this csv file, all aisles are present associated with their aisle id. Contains 134 rows and 2 columns.

2. departments.csv: In this csv file, all departments are present associated with their department id. Contains 121 rows and 2 columns.

3. order_products__prior.csv: This file holds order_id data associated with the n number product_ids and also other data. Contains 32434489 and 4 columns.

4. orders.csv: This csv file holds user_ids and their order_ids. Additional info includes buying date, buying hour and days since prior order. Contains 3421083 rows and 7 columns

5. products.csv: In this csv file, all products are present associated with their product id. Contains 49688 rows and 4 columns.

<img src="Instacart Dataset's ERD Diagram.png"/>

**The Hierarchy present in the dataset**

There is a hierarchy present in the dataset which must be understood to truly understand the dataset. The hierarchy is as follows:

At the top there are departments and then there are aisles and at the bottom there are products.

A single department has multiple aisles and each of these aisles have multiple products.